pkg/resources/dnsrecord/dns.go

```diff
@@ -27,58 +32,102 @@ import (
 // [1] https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-choosing-alias-non-alias.html
 const defaultRecordTTL int64 = 30
 
-// ensureWildcardDNSRecord will create DNS records for the given LB service.
-// If service is nil (haveLBS is false), nothing is done.
-func (r *reconciler) ensureWildcardDNSRecord(ic *operatorv1.IngressController, service *corev1.Service, haveLBS bool) (bool, *iov1.DNSRecord, error) {
+// EnsureWildcardDNSRecord will create wildcard DNS records for the given LB
+// service.  If service is nil (haveLBS is false), nothing is done.
+func EnsureWildcardDNSRecord(client client.Client, name types.NamespacedName, dnsRecordLabels map[string]string, ownerRef metav1.OwnerReference, domain string, endpointPublishingStrategy *operatorv1.EndpointPublishingStrategy, service *corev1.Service, haveLBS bool) (bool, *iov1.DNSRecord, error) {
 	if !haveLBS {
 		return false, nil, nil
 	}
 
-	wantWC, desired := desiredWildcardDNSRecord(ic, service)
-	haveWC, current, err := r.currentWildcardDNSRecord(ic)
+	wantWC, desired := desiredWildcardDNSRecord(name, dnsRecordLabels, ownerRef, domain, endpointPublishingStrategy, service)
+	haveWC, current, err := CurrentDNSRecord(client, name)
 	if err != nil {
 		return false, nil, err
 	}
 
 	switch {
 	case wantWC && !haveWC:
-		if err := r.client.Create(context.TODO(), desired); err != nil {
+		if err := client.Create(context.TODO(), desired); err != nil {
 			return false, nil, fmt.Errorf("failed to create dnsrecord %s/%s: %v", desired.Namespace, desired.Name, err)
 		}
 		log.Info("created dnsrecord", "dnsrecord", desired)
-		return r.currentWildcardDNSRecord(ic)
+		return CurrentDNSRecord(client, name)
 	case wantWC && haveWC:
-		if updated, err := r.updateDNSRecord(current, desired); err != nil {
+		if updated, err := updateDNSRecord(client, current, desired); err != nil {
 			return true, current, fmt.Errorf("failed to update dnsrecord %s/%s: %v", desired.Namespace, desired.Name, err)
 		} else if updated {
-			return r.currentWildcardDNSRecord(ic)
+			return CurrentDNSRecord(client, name)
+		}
+	}
+
+	return haveWC, current, nil
+}
+
+// EnsureDNSRecord will create DNS records for the given LB service.  If service
+// is nil (haveLBS is false), nothing is done.
+func EnsureDNSRecord(client client.Client, name types.NamespacedName, dnsRecordLabels map[string]string, ownerRef metav1.OwnerReference, domain string, service *corev1.Service) (bool, *iov1.DNSRecord, error) {
```

Comment:
Is there a better way to structure this code opposed to having separate functions for `EnsureDNSRecord` and `EnsureWildCardDNSRecord`? It feels a little repetitive to me.

Reply:
I could move the wildcard prefixing logic to the place that calls `EnsureWildCardDNSRecord`, but my goal was to avoid changing that code more than necessary.

Reply:
Makes sense, we can refactor later to prevent more churn.